osf/models/node.py

```diff
@@ -2125,6 +2126,12 @@ def update(self, fields, auth=None, save=True):
             elif key == 'category':
                 self.set_category(category=value, auth=auth, save=False)
             elif key == 'is_public':
+                if value is False:
```

Comment:
nit: I'd argue that it makes more sense for this to get added to the set_privacy function (the "is public, being made private" case is already here https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/node.py#L1229)

Reply:
I opted to add this logic to the `update` method because `set_privacy` is being used both for spam (which should be modified regardless of storage caps) and for over-riding storage caps when modifying a node's public status. We could modify `set_privacy` to have some type of `override storage caps` parameter but would then have to update the calls to set_privacy accordingly. Do you think having that type of override parameter would be better than having the check in `update`?

Reply:
If you were to move this to `set_privacy`, you could check for `auth=None`, which indicates that it's being modified by the admin app rather than a user. That would be an alternative to an explicit flag.

Reply:
Making the set_privacy signature more convoluted is probably not worth it, but if Brian's suggestion works, that would be great. This just looks like the only key that's is going through this level of business logic in this function.